packages/fetchai/connections/p2p_libp2p/check_dependencies.py

```diff
@@ -215,10 +215,42 @@ def _golang_module_build(
     return None
 
 
+def search_for_source_code() -> str:
+    """
+    Search for the source directory.
+
+    We search along the whole path to the file, looking into potential directories along it.
+    """
+    path = str(os.path.abspath(os.path.dirname(__file__)))
+    FILE = "libp2p_node.go"
+    POTENTIAL_DIRS = [str(Path("libs", "go", "libp2p_node")), "libp2p_node"]
+    source_code_dir = ""
+    is_root = False
+    done = False
+    while not is_root and not done:
+        path = str(Path(path).parent)
+        for dir_ in POTENTIAL_DIRS:
+            full_path = str(Path(path, dir_, FILE))
+            if os.path.isfile(full_path):
+                source_code_dir = str(Path(path, dir_))
+                done = True
+        is_root = path == "."
+    return source_code_dir
+
+
 def build_node(build_dir: str) -> None:
     """Build node placed inside build_dir."""
+    GITHUB_URL = "https://github.com/fetchai/agents-aea/tree/main/libs/go/libp2p_node"
+    source_code_dir = search_for_source_code()
```

Comment:
We first search along path, so to deal with development and also to give the option to just download the folder and place it in the AEA project.